tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -288,27 +316,66 @@ class UnsortedSegmentSumOp : public OpKernel {
     auto output_flat = output->flat_outer_dims<T>();
 
     auto data_ptr = data.template flat<T>().data();
-    functor::UnsortedSegmentSumFunctor<Device, T, Index>()(
-        context, context->template eigen_device<Device>(), output_rows,
+//      this->operator()(context, context->template eigen_device<Device>(), output_rows,
+//        segment_ids.shape(), segment_flat, data.NumElements(), data_ptr,
+//        output_flat);
+      f_->operator()(context, context->template eigen_device<Device>(), output_rows,
         segment_ids.shape(), segment_flat, data.NumElements(), data_ptr,
         output_flat);
   }
+  functor::UnsortedSegmentBaseFunctor< Device, T, Index>* f_;
 };
 
-#define REGISTER_CPU_UNSORTED_KERNELS(type, index_type)                \
+template <typename Device, class T, class Index>
```

Comment:
This doesn't look right. I think we need continuation for all lines. I'm not even that builds?